compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -366,6 +411,128 @@ void LinalgFusePass::runOnOperation() {
+    // Make sure that only generic ops are supported, only one input and the
+    // innermost tensor dimension is reduction dimension.  Everything else is
+    // copied from the original SplitReduction Transformation.
+    if (failed(filter.checkAndNotify(rewriter, op)) ||
+        !op.hasTensorSemantics() || op.getNumReductionLoops() != 1 ||
+        op.getNumOutputs() != 1 || !op.hasOnlyProjectedPermutations() ||
+        !isa<linalg::GenericOp>(op) || op.getNumInputs() != 1)
+      return rewriter.notifyMatchFailure(op, "precondition not met");
+    SmallVector<unsigned> dims;
+    op.getReductionDims(dims);
+    AffineMap map = op.getTiedIndexingMap(op.getInputOperand(0));
+    unsigned lastIdx = map.getNumResults() - 1;
+    unsigned lastDim = map.getDimPosition(lastIdx);
+    if (lastDim != dims[0])
+      return rewriter.notifyMatchFailure(op, "precondition not met");
+
+    linalg::ControlSplitReductionFn fn = [this, lastIdx](linalg::LinalgOp op) {
+      return linalg::SplitReductionOptions{size, lastIdx, true};
+    };
+
+    auto opNumLoops = op.getNumLoops();
+
+    // 1) Tile to extract a single vector-length array.
+    SmallVector<int64_t> tileSizesSVFirst(opNumLoops, 1);
+    tileSizesSVFirst[opNumLoops - 1] = 0;
+    ArrayRef<int64_t> tileSizesFirst(tileSizesSVFirst);
+    linalg::LinalgTilingOptions optionsFirst;
+    optionsFirst.setTileSizes(tileSizesFirst);
+    FailureOr<linalg::TiledLinalgOp> tileResFirst =
+        linalg::tileLinalgOp(rewriter, op, optionsFirst);
+    if (failed(tileResFirst)) return failure();
+    rewriter.replaceOp(op, tileResFirst->tensorResults);
+    filter.replaceLinalgTransformationFilter(rewriter, tileResFirst->op);
+
+    // 2) Apply splitReduction on the single vector-length array. splitReduction
+    // already replaces the op.
+    FailureOr<linalg::SplitReductionResult> splitRes =
+        splitReduction(rewriter, tileResFirst->op, fn);
+    if (failed(splitRes)) return failure();
+    filter.replaceLinalgTransformationFilter(rewriter, splitRes->splitLinalgOp);
+    filter.replaceLinalgTransformationFilter(rewriter,
+                                             splitRes->resultCombiningLinalgOp);
+
+    // 3) Tile the first op generated by splitReduction with tile size of 1, to
+    // essentially create a reduction loop.
+    SmallVector<int64_t> tileSizesSV(opNumLoops + 1, 0);
+    tileSizesSV[opNumLoops - 1] = 1;
+    ArrayRef<int64_t> tileSizes(tileSizesSV);
+    linalg::LinalgTilingOptions options;
+    options.setTileSizes(tileSizes);
+    FailureOr<linalg::TiledLinalgOp> tileRes =
+        linalg::tileLinalgOp(rewriter, splitRes->splitLinalgOp, options);
+    if (failed(tileRes)) return failure();
+    rewriter.replaceOp(splitRes->splitLinalgOp, tileRes->tensorResults);
+    filter.replaceLinalgTransformationFilter(rewriter, tileRes->loops[0]);
+
+    // // 4) Unroll the tiled op (scf.for) by unrollFactor. loopUnrollByFactor
```

Comment:
Remove commented code...

Reply:
That code is an integral part of the PR, except that it crashes now (probably because `loopUnrollByFactor` is buggy). I would rather fix the bug in this PR if it's not too tedious.

Reply:
In this case, let's replace them with descriptive comments?